src/mjolnir/graphbuilder.cc

```diff
@@ -53,13 +53,15 @@ std::map<GraphId, size_t> SortGraph(const std::string& nodes_file, const std::st
   // to point to the first (out of the duplicates) nodes index. at the end of this there will be
   // tons of nodes that no edges reference, but we need them because they are the means by which
   // we know what edges connect to a given node from the nodes perspective
-  sequence<Edge> edges(edges_file, false);
+
   uint32_t run_index = 0;
   uint32_t node_index = 0;
   size_t node_count = 0;
   Node last_node{};
   std::map<GraphId, size_t> tiles;
-  nodes.transform([&edges, &run_index, &node_index, &node_count, &last_node, &tiles](Node& node) {
+  // If these get too large, consider using a sequence instead of a vector here.
+  std::vector<std::pair<uint32_t, uint32_t>> starts, ends;
```

Comment:
at 8 bytes per record and about 400000000 graph edges (double that for because of start/end) we get something like 3.2GB (and maybe double that again if we are unlucky with vector allocation). this is definitely close to needing a sequence. would you be opposed to something as simple as a struct (right here above the function) to encapsulate the two indices and then throwing them into a sequence?

Reply:
Yep, I can do that!

Reply:
I think it might be easier to have two separate sequences; otherwise you need an extra bit of data for each entry indicating whether it's a start or an end. If instead the sequence was a mapping [edge -> (start,end)], you run into the same random insert issue as you iterate through the nodes.

Reply:
ah yep sorry i was being imprecise with my wording above. i meant 2 sequences in total, 1 for each mapping.